src/Core/Domain/Product/QueryResult/ProductType.php

```diff
@@ -67,35 +67,35 @@ class ProductType
     ];
 
     /**
-     * @var int
+     * @var string
      */
     private $value;
 
     /**
-     * @param int $value
+     * @param string $value
      *
      * @throws ProductConstraintException
      */
-    public function __construct(int $value)
+    public function __construct(string $value)
     {
         $this->assertProductType($value);
         $this->value = $value;
     }
 
     /**
-     * @return int
+     * @return string
      */
-    public function getValue(): int
+    public function getValue(): string
     {
         return $this->value;
     }
 
     /**
-     * @param int $value
+     * @param string $value
      *
      * @throws ProductConstraintException
      */
-    private function assertProductType(int $value): void
+    private function assertProductType(string $value): void
```

Comment:
I believe this checking part and especially the exception throwing should be removed
In query result I don't think we throw exceptions because we are supposed to control the data provided so it shouldn't be needed (not like in ValueObject which contains input from user)

Reply:
I get the point. But you cannot avoid anyone to instantiate an object from anywhere. I guess this is a miniaml security to at least check that expected types are used